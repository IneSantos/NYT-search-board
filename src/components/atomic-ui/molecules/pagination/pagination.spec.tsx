import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { render } from '@testing-library/react';
import Pagination from "./pagination";

let store;
let mockedStore;

describe("Pagination component tests", () => {
  test(`Pagination component render`, () => {
    mockedStore = configureStore();
    store = mockedStore({  
        articleList: [{
            id: "",
            headline: "",
            abstract: "", 
            keywords: [],
            section: "",
            snippet: "",
            pub_date: "",
            source: "",
            web_url: ""
        }],
        page: 0,
        searchClicked: false
    });


    const { container } = render(
        <Provider store={store}>
            <Pagination/>
         </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
