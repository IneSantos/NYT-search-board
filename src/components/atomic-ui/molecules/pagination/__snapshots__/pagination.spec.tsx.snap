// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component tests Pagination component render 1`] = `
<div
  class="pagination"
>
  <a
    class="article-action article-action-next-page"
  >
     
    Next page &gt;
     
  </a>
  <a
    class="article-action article-action-prev-page"
  >
     
    &lt; Prev page
     
  </a>
</div>
`;
